lib/cadet_web/plug/check_admin.ex

```diff
@@ -0,0 +1,21 @@
+defmodule CadetWeb.Plug.CheckAdmin do
+  @moduledoc """
+  Checks whether :current_user is an admin.
+  If the user is not an admin, the HTTP 403 response will be sent.
+  """
+
+  import Plug.Conn
+
+  def init(opts), do: opts
+
+  def call(conn, _) do
+    if conn.assigns[:current_user].role == :admin do
```

Comment:
Also I just realised this is wrong, the admin interface is meant to be opened by both Admin and Staff. So `EnsureStaff` will do